### PR TITLE
Update input context in DoEvents instead of DoUpdate

### PR DESCRIPTION
**Pull request by Perksey**
As discussed in Discord

### CODE REVIEW

File: src/Input/Silk.NET.Input.Glfw/GlfwInputContext.cs [hunk 1]
```diff
@@ -50,30 +50,48 @@ public unsafe GlfwInputContext(IView window)
...
         public unsafe void Dispose()
         {
-            _window.Update -= _update;
+            if (_window is ViewImplementationBase view)
+            {
+                view.ProcessEvents += ProcessEvents;
+            }
+            else
+            {
+                _window.Update += Update;
```
Comment (vdvman1) [inline src/Input/Silk.NET.Input.Glfw/GlfwInputContext.cs:92]: Shouldn't these be removing the event handlers rather than adding them again?

Comment (Perksey) [reply src/Input/Silk.NET.Input.Glfw/GlfwInputContext.cs:92]: Yes, thanks!

```suggestion
                _window.Update -= Update;
```

File: src/Input/Silk.NET.Input.Glfw/GlfwInputContext.cs [hunk 2]
```diff
@@ -50,30 +50,48 @@ public unsafe GlfwInputContext(IView window)
...
+                updatable.Update();
+            }
+        }
+
         public unsafe void Dispose()
         {
-            _window.Update -= _update;
+            if (_window is ViewImplementationBase view)
+            {
+                view.ProcessEvents += ProcessEvents;
```
Comment (Perksey) [inline src/Input/Silk.NET.Input.Glfw/GlfwInputContext.cs:88]:
```suggestion
                view.ProcessEvents -= ProcessEvents;
```

**Comment by ThomasMiz** [comment]
Shouldn't we change SDL to also use the new `ProcessEvents` event? It seems to basically be a generalized version of [this](https://github.com/dotnet/Silk.NET/blob/3a5f4508c62eaf6eb3d2466fb971ebc2b359ce12/src/Windowing/Silk.NET.Windowing.Sdl/SdlView.cs#L96) so I don't see why not.

Would be extremely confusing if `SdlView` had both the inherited `ProcessEvents` and a separate `ProcessingEvents` event

**Comment by Perksey** [comment]
You're right, I've changed this to make the same event common (and make other code common as well)

**Review by ThomasMiz — APPROVED** [review]

**Pull request merged**